contracts/hackatom/tests/integration.rs

```diff
@@ -126,32 +126,42 @@ fn fails_on_bad_init() {
 }
 
 #[test]
-fn proper_handle() {
-    let mut deps = mock_instance(WASM, &coins(1015, "earth"));
+fn handle_release_works() {
+    let mut deps = mock_instance(WASM, &[]);
 
     // initialize the store
-    let verifier = HumanAddr(String::from("verifies"));
-    let beneficiary = HumanAddr(String::from("benefits"));
+    let creator = HumanAddr::from("creator");
+    let verifier = HumanAddr::from("verifies");
+    let beneficiary = HumanAddr::from("benefits");
 
     let init_msg = InitMsg {
         verifier: verifier.clone(),
         beneficiary: beneficiary.clone(),
     };
-    let init_env = mock_env(&deps.api, "creator", &coins(1000, "earth"));
+    let init_amount = coins(1000, "earth");
+    let init_env = mock_env(&deps.api, creator.as_str(), &init_amount);
+    let contract_addr = deps.api.human_address(&init_env.contract.address).unwrap();
     let init_res: InitResponse = init(&mut deps, init_env, init_msg).unwrap();
-    assert_eq!(0, init_res.messages.len());
+    assert_eq!(init_res.messages.len(), 0);
+
+    // balance changed in init
+    deps.with_querier(|querier| {
```

Comment:
Nice update function. This will simplify a lot of the staking code as well where I need to access the backing querier for the test cases.